File: Python/349__Intersection_of_Two_Arrays.py
class Solution(object):
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        dic = dict()
        result = []
        for i in range(len(nums1)) :
            if nums1[i] not in dic :
                dic[nums1[i]] = 1
        for i in range(len(nums2)) :
            if nums2[i] in dic and nums2[i] not in result :
                result += [nums2[i]]
        return result

File: Java/349__Intersection_of_Two_Arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> set = new HashSet<>();
        for (int val : nums1) {
            set.add(val);
        }
        ArrayList<Integer> result_array = new ArrayList<>();
        for (int val : nums2) {
            if (!result_array.contains(val) && set.contains(val)) {
                result_array.add(val);
            }
        }
        int[] result = new int[result_array.size()];
        int i = 0;
        for (int val : result_array) {
            result[i] = val;
            i++;
        }
        return result;
    }
}

// Similar Solution
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        Set<Integer> set = new HashSet<>();
        Set<Integer> result_set = new HashSet<>();
        for (int num :nums1) {
            set.add(num);
        }
        for (int num :nums2) {
            if (set.contains(num)) {
              result_set.add(num);
            }
        }
        int [] result = new int [result_set.size()];
        int i = 0;
        for (int val : result_set){
            result[i++] = val;
        }
        return result;
    }
}
